core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -198,9 +205,11 @@ public ChannelImpl shutdown() {
    *
    * <p>NOT YET IMPLEMENTED. This method currently behaves identically to shutdown().
    */
-  // TODO(ejona86): cancel preexisting calls.
   public ChannelImpl shutdownNow() {
     shutdown();
+    for (ClientCall<?, ?> call : activeCalls) {
+      call.cancel();
```

Comment:
It cancels the calls, but it doesn't say "cancel is called on each call". The cancel on Call isn't thread-safe anyway; don't use it here.